data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -266,12 +254,15 @@ public void testRandomData() throws IOException {
     append.commit();
 
     Set<Record> records = Sets.newHashSet(IcebergGenerics.read(table).build());
-    Assert.assertEquals(
-        "Should produce correct number of records", expected.size(), records.size());
-    Assert.assertEquals("Random record set should match", Sets.newHashSet(expected), records);
+    Assertions.assertThat(expected)
+        .hasSize(records.size())
+        .as("Should produce correct number of records");
```

Comment:
`.as(..)` needs to come before the final assertion. Also the check should be `assertThat(records).as(..).hasSameSizeAs(expected)`